From 059e95ae6b8c4843829cb901b863853199f1c88d Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Thu, 27 Sep 2018 12:15:06 +0200
Subject: [PATCH 5/6] sdk: Update default clock position

* Also migrate users using right clock so
  that they get the new default setting.

Change-Id: I2e38a16af46d77ff92ada268b8fc0927b0c4806c

# Conflicts:
#	packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java
---
 .../res/values/defaults.xml                   |  2 +-
 .../LineageDatabaseHelper.java                | 21 ++++++++++++++++++-
 2 files changed, 21 insertions(+), 2 deletions(-)

diff --git a/packages/LineageSettingsProvider/res/values/defaults.xml b/packages/LineageSettingsProvider/res/values/defaults.xml
index 1c4a027..3a23f52 100644
--- a/packages/LineageSettingsProvider/res/values/defaults.xml
+++ b/packages/LineageSettingsProvider/res/values/defaults.xml
@@ -118,6 +118,6 @@
          * 1: show the clock in the center
          * 2: show the clock in the left position (LTR)
     -->
-    <integer name="def_clock_position">0</integer>
+    <integer name="def_clock_position">2</integer>
 
 </resources>
diff --git a/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java b/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java
index 22af6aa..b15a264 100644
--- a/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java
+++ b/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java
@@ -336,8 +336,27 @@ public class LineageDatabaseHelper extends SQLiteOpenHelper{
                     db.setTransactionSuccessful();
                 } finally {
                     if (stmt != null) stmt.close();
-                    db.endTransaction();
                 }
+
+                stmt = null;
+                try {
+                    stmt = db.compileStatement("SELECT value FROM system WHERE name=?");
+                    stmt.bindString(1, LineageSettings.System.STATUS_BAR_CLOCK);
+                    long value = stmt.simpleQueryForLong();
+
+                    if (value == 0) {
+                        stmt = db.compileStatement("UPDATE system SET value=? WHERE name=?");
+                        stmt.bindLong(1, 2);
+                        stmt.bindString(2, LineageSettings.System.STATUS_BAR_CLOCK);
+                        stmt.execute();
+                    }
+                    db.setTransactionSuccessful();
+                } catch (SQLiteDoneException ex) {
+                    // LineageSettings.System.STATUS_BAR_CLOCK is not set
+                } finally {
+                    if (stmt != null) stmt.close();
+                }
+                db.endTransaction();
             }
             upgradeVersion = 10;
         }
-- 
2.17.1

